### PR TITLE
Adapter-static documentation: Github pages

### PR DESCRIPTION
**Pull request by Marr11317**
Add documentation for publishing to Github pages with adapter-static.
Would an `adapter-gh-pages` be possible?

### CODE REVIEW

File: packages/adapter-static/README.md
Comment (benmccann) [inline packages/adapter-static/README.md:68]:
```suggestion
Also note that you will have to prevent Github's provided Jekyll from managing your site by putting an empty `.nojekyll` file in your static folder. If you do not want to disable Jekyll, change the kit's `appdir` configuration option to `"app_"` or anything not starting by an underscore. For more information, see Github's documentation about [configuring jekyll](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#configuring-jekyll-in-your-github-pages-site).
```

File: packages/adapter-static/README.md
Comment (jthegedus) [inline packages/adapter-static/README.md:66]: Perhaps an example valid config for GitHub pages?

**Comment by Marr11317** [comment]
Example added

File: packages/adapter-static/README.md
Comment (benmccann) [inline packages/adapter-static/README.md:80]: This seems like quite the caveat. I don't know if we'd want to promote this solution without making it a bit nicer to use first

Comment (benmccann) [reply packages/adapter-static/README.md:80]: paths are currently broken in dev mode. i'll try to review https://github.com/sveltejs/kit/pull/2145 and then we can see if that changes the situation

File: packages/adapter-static/README.md
Comment (Rich-Harris) [inline packages/adapter-static/README.md:64]: hate to be a cop but we should fix all these

```suggestion
## GitHub Pages
```

File: packages/adapter-static/README.md
Comment (Rich-Harris) [inline packages/adapter-static/README.md:68]:
```suggestion
You will have to prevent GitHub's provided Jekyll from managing your site by putting an empty `.nojekyll` file in your static folder. If you do not want to disable Jekyll, change the kit's `appDir` configuration option to `"app_"` or anything not starting with an underscore. For more information, see GitHub's [Jekyll documentation](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/about-github-pages-and-jekyll#configuring-jekyll-in-your-github-pages-site).
```

File: packages/adapter-static/README.md
Comment (Rich-Harris) [inline packages/adapter-static/README.md:70]:
```suggestion
A config for Github Pages might look like the following:
```

File: packages/adapter-static/README.md
Comment (Rich-Harris) [inline packages/adapter-static/README.md:79]:
```suggestion
		// Since your GitHub Pages site will be served at https://your-username.github.io/your-repo-name,
		// you will need to change the base paths
```

File: packages/adapter-static/README.md
Comment (Rich-Harris) [inline packages/adapter-static/README.md:83]: `assets` is resolved relative to `base`, so it can be left as the default

```suggestion
			base: '/your-repo-name'
```

File: packages/adapter-static/README.md
Comment (Rich-Harris) [inline packages/adapter-static/README.md:87]:
```suggestion
		appDir: 'internal'
```

**Review by Rich-Harris — CHANGES_REQUESTED** [review]

File: packages/adapter-static/README.md
Comment (benmccann) [inline packages/adapter-static/README.md:86]:
```suggestion
		// For example, instead of '_app', use 'app_', 'internal', etc.
```

File: packages/adapter-static/README.md
Comment (Marr11317) [reply packages/adapter-static/README.md:80]: other option is to detect the dev environment: 
```js
const isProduction = process.env.NODE_ENV === 'production';

...
    paths: {
        base: isProduction ? "project_name" : "",
    }
...
```

**Comment by Rich-Harris** [comment]
I tweaked the language to remove that caveat — we can merge this once #2189 is merged

**Review by benmccann — APPROVED** [review]

**Pull request merged**
